# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/cache/Manager.cpp
# Commit: c8a45415dc5b
# Full Hash: c8a45415dc5b0b33b7e9d4b9cd8114ac09a05218
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-05-28 21:48:03
# Description:
#   Bug 1706998: Replace QuotaManager::GetRef().MaybeRecordShutdownStep by static QuotaManager::(Safe)MaybeRecordQuotaClientShutdownStep; r=dom-storage-reviewers,janv
#   
#   There are some few unsafe uses of MaybeRecordShutdownStep where the QuotaManager singleton may be  not (yet or anymore) alive.
#   In order to not add many unnecessary null checks, we drop GetRef and MaybeRecordShutdownStep in favor of
#   QuotaManager::MaybeRecordQuotaClientShutdownStep
# ==============================================================================

diff -r cc21fb9be25a -r c8a45415dc5b dom/cache/Manager.cpp
--- a/dom/cache/Manager.cpp	Fri May 28 05:49:26 2021 +0000
+++ b/dom/cache/Manager.cpp	Fri May 28 06:46:04 2021 +0000
@@ -267,7 +267,10 @@
 
     MOZ_ALWAYS_TRUE(sFactory->mManagerList.RemoveElement(&aManager));
 
-    quota::QuotaManager::GetRef().MaybeRecordShutdownStep(
+    // This might both happen in late shutdown such that this event
+    // is executed even after the QuotaManager singleton passed away
+    // or if the QuotaManager has not yet been created.
+    quota::QuotaManager::SafeMaybeRecordQuotaClientShutdownStep(
         quota::Client::DOMCACHE, "Manager removed"_ns);
 
     // clean up the factory singleton if there are no more managers